src/transformStateWithClones.js

```diff
@@ -6,8 +6,35 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = Object.assign({}, state);
+
+  for (const action of actions) {
+    let newState = Object.assign({}, currentState);
```

Comment:
[NAMING] - The variable name 'newState' could be improved to better reflect its purpose. Consider naming it 'stateCopy' or similar to indicate that it is a copy of the current state.